### PR TITLE
Add create new/use existing virtual network

### PR DESCRIPTION
**Pull request by russcam**
Add virtual network settings blade to UI definition
Add virtual network settings as parameters to deployment template

Use Start IP address of the master subnet as the static IP of internal load balancer
Set default url for Sense plugin to the internal load balancer IP so populated with correct url on first time.

Use the last 8 characters of the cluster name as a namespace prefix for the virtual machine names.
- If the cluster name is less than 8 characters, left pad with *.
- Names are used in the elasticsearch install script for unicast discovery.

Dynamic IP addresses for master nodes in master subnet

Consistent namespaces across master, data, client and jumpbox resources
- Names are concatenated in the main template

Fix error with output from kibana template when not specifying to install Kibana

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ The market place Elasticsearch offering offers a simplified UI over the full pow
...
+You can view the UI in developer mode by [clicking here](https://portal.azure.com/#blade/Microsoft_Azure_Compute/CreateMultiVmWizardBlade/internal_bladeCallId/anything/internal_bladeCallerParams/{"initialData":{},"providerConfig":{"createUiDefinition":"https%3A%2F%2Fraw.githubusercontent.com%2Felastic%2Fazure-marketplace%2Fhackfest%2Fsrc%2FcreateUiDefinition.json"}}). If you feel something is cached improperly use [this client unoptimized link instead](https://portal.azure.com/?clientOptimizations=false#blade/Microsoft_Azure_Compute/CreateMultiVmWizardBlade/internal_bladeCallId/anything/internal_bladeCallerParams/{"initialData":{},"providerConfig":{"createUiDefinition":"https%3A%2F%2Fraw.githubusercontent.com%2Felastic%2Fazure-marketplace%2Fhackfest%2Fsrc%2FcreateUiDefinition.json"}})
```
Comment (Mpdreamz) [inline README.md]: the `hackfest` links need to be switched back to `master`

**Comment by ajhewett** [comment]
The feature to use an existing vnet would be a very welcome improvement.  I want to use a vnet containing a vpn gateway so that all access to the ES cluster goes through the vpn and an internal load balancer.  A-postori adding a vpn gateway to the vnet causes problems when re-deploying the ES template (e.g. to increase the number of data nodes).

**Comment by russcam** [comment]
Updated this PR following merge of the latest template release.

- Add virtual network settings to UI definition in cluster settings step (reduce number of blades).
- Add virtual network settings as parameters to deployment template
- Ask for static IP address in the subnet for the internal load balancer
- Set default url for Sense plugin to the internal load balancer IP so populated with correct url on first time.
- Ask for hostname prefix to use for the virtual machine names/hostnames. This can be left blank, but it recommended to set if deploying a cluster into an existing virtual network that has a deployed cluster in it. This is to avoid hostname resolution conflicts.
- Consistent namespaces across master, data, client and jumpbox resources; all use the hostname prefix.
- Names are concatenated in the main template
- Fix error with output from kibana template when not specifying to install Kibana

**Comment by Mpdreamz** [comment]
> Add virtual network settings to UI definition in cluster settings step (reduce number of blades).

For now i think this is good, when we tackle #34 we should move the cluster name and elasticsearch version to "Nodes" and rename it "Topology" and rename this blade "Network". <-- just a suggestion but we might need to rethink the overal flow as part of #34. ++ for not introducing another blade now :+1:

**Comment by russcam** [comment]
#34 is next item on the list :+1:

File: build/arm-tests/3d-0m-3c-ext-kp-location.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
 	"why" : "",
 	"location" : "westeurope",
 	"parameters" : {
-		"loadBalancerType":{"value":"external"},
+		"loadBalancerType":{"value":"External"},
```
Comment (Mpdreamz) [inline build/arm-tests/3d-0m-3c-ext-kp-location.json]: I sort of agree with change but it invalidates all existing arm automation for little gains IMO, i rather we reverse this back to `internal` and `external`

Comment (russcam) [inline build/arm-tests/3d-0m-3c-ext-kp-location.json]: I was thinking along the lines of consistency; we have `Yes/No` options for kibana and loadbalancer, but then `Internal/Eternal` for load balancer type and `new/existing` for new or existing virtual network. 

Can we lowercase all of the options?

**Pull request merged**